src/Utilities/MakeVector.hpp

```diff
@@ -0,0 +1,37 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <vector>
+
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+/*!
+ * \brief Constructs a `std::vector` containing 'arg_1' and each of the
```

Comment:
Why these details in the brief? How about, `Constructs a std::vector containing each argument passed in`.